pkg/operation/botanist/waiter.go

```diff
@@ -83,8 +85,13 @@ func (b *Botanist) WaitUntilNoPodRunning(ctx context.Context) error {
 		for _, pod := range podList.Items {
 			if pod.Status.Phase == corev1.PodRunning {
 				b.Logger.Info("Waiting until there are no running pods in the shoot cluster (at least one pod still exists)", "pod", client.ObjectKeyFromObject(&pod))
-				return retry.MinorError(fmt.Errorf("waiting until there are no running Pods in the shoot cluster... "+
-					"there is still at least one running Pod in the shoot cluster: %q", client.ObjectKeyFromObject(&pod).String()))
+
+				message := "waiting until there are no running Pods in the shoot cluster ... there is still at least one running Pod in the shoot cluster: %q"
+				if pod.Namespace == metav1.NamespaceSystem {
+					return retry.MinorError(fmt.Errorf(message, client.ObjectKeyFromObject(&pod).String()))
+				} else {
+					return retry.MinorError(helper.NewErrorWithCodes(fmt.Errorf(message, client.ObjectKeyFromObject(&pod).String()), gardencorev1beta1.ErrorCleanupClusterResources))
+				}
```

Comment:
Nit: make it more concise.

```suggestion
				}
				return retry.MinorError(helper.NewErrorWithCodes(fmt.Errorf(message, client.ObjectKeyFromObject(&pod).String()), gardencorev1beta1.ErrorCleanupClusterResources))
```

Reply:
Why did you resolve this when you did not apply it or comment?

Reply:
https://github.com/gardener/gardener/pull/9060/commits/b3692100e2b0fadc2e99ac5c3c78512f459d7c3a, should fix the mentioned issue, therefore I resolved the comment.

Reply:
No, it does not. The `else` is not required as the `if` block always `return`s. Therefore, you can convert the `else` block into a simple `return` statement:

```
if ... {
    return ...
}
return ...
```

This is what the suggestion was all about and it was not addressed in https://github.com/gardener/gardener/commit/b3692100e2b0fadc2e99ac5c3c78512f459d7c3a.

Reply:
Apologies, I misunderstood your comment. Should be fixed now, but feel free to check and then close.